actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/BatchOperationNotifierActor.java

```diff
@@ -0,0 +1,421 @@
+   */
+  private void sendMail(Map<String, String> user, Map<String, Object> requestMap) {
+    String template = getEmailTemplateFile(user.get("templateName"));
+    requestMap.put(JsonKey.FIRST_NAME, user.get(JsonKey.FIRST_NAME));
+    VelocityContext context = getContext(requestMap);
+    String decryptedEmail = decryptionService.decryptData(user.get(JsonKey.EMAIL));
+    if (decryptedEmail != null && !StringUtils.isBlank(decryptedEmail)) {
+      try {
+        SendMail.sendMailWithBody(
+            new String[] {decryptedEmail}, user.get(JsonKey.SUBJECT), context, template);
+      } catch (Exception e) {
+        ProjectLogger.log(
+            "BatchOperationNotifierActor : sendMail, Error encountered while sending email notification for batch operation for user   "
+                + e.getMessage(),
+            LoggerEnum.ERROR.name());
+      }
+    } else {
+      ProjectLogger.log(
+          "BatchOperationNotifierActor : sendMail, Error encountered while sending email, eitheer it is black or null ",
+          LoggerEnum.ERROR.name());
+    }
+  }
+
+  /*
+   * This method takes CourseBatch object and returns list of participants
+   *
+   * @param CourseBatch object
+   *
+   * @returns list of users
+   */
+  private List<String> getParticipants(CourseBatch courseBatch) {
+    List<String> usersIds = null;
+    Map<String, Boolean> participants = courseBatch.getParticipant();
+    if (participants != null) {
+      usersIds = new ArrayList<>();
+      Set<String> keys = participants.keySet();
+      for (String user : keys) {
+        if (participants.get(user)) {
+          usersIds.add(user);
+        }
+      }
+    }
+    return usersIds;
+  }
+
+  /*
+   * This method takes List<String>userIds and returns List<Map<User,Objects>> of
+   * participants
+   *
+   * @param List<String> userIds list of userIds
+   *
+   * @return List<Map<String, Object>> map of user related data.
+   */
+  private List<Map<String, Object>> getUsersFromDB(List<String> userIds) {
+    Util.DbInfo usrDbInfo = Util.dbInfoMap.get(JsonKey.USER_DB);
+    List<String> userIdList = new ArrayList<>(userIds);
+    List<String> fields = new ArrayList<>();
+    fields.add(JsonKey.FIRST_NAME);
+    fields.add(JsonKey.EMAIL);
+    Response response =
+        cassandraOperation.getRecordsByIdsWithSpecifiedColumns(
+            usrDbInfo.getKeySpace(), usrDbInfo.getTableName(), fields, userIdList);
+    if (response == null) {
+      ProjectLogger.log("No data from cassandra , check connection  ", LoggerEnum.ERROR.name());
+    }
+    List<Map<String, Object>> userList = (List<Map<String, Object>>) response.get(JsonKey.RESPONSE);
+    return userList;
+  }
+
+  /*
+   * This method takes template name and return template.
+   *
+   * @param templateName String
+   *
+   * @return email template String .
+   */
+  private String getEmailTemplateFile(String templateName) {
+    EmailTemplateDao emailTemplateDao = EmailTemplateDaoImpl.getInstance();
```

Comment:
The notification module will use EmailTemplateDao. This class shouldn't directly access email template table.